src/agent/rustjail/src/sync.rs

```diff
@@ -72,7 +72,15 @@ fn read_count(fd: RawFd, count: usize) -> Result<Vec<u8>> {
         }
     }
 
-    Ok(v[0..len].to_vec())
+    if len != count {
```

Comment:
This is supposed to duplicate https://doc.rust-lang.org/std/io/trait.Read.html#method.read_exact ? What about just using the std one?

Reply:
Hi @bergwolf , if we use the std one here, it'll  need to do the interchange between raw fd  and std::fs::File which would make this fn dirty. 
I'll refactor those sync functions in a separated PR and replace it with std one.

Reply:
sounds good to me!